WordPress/src/main/java/org/wordpress/android/util/SiteUtils.java

```diff
@@ -103,6 +104,15 @@ public static void migrateAppWideMobileEditorPreferenceToRemote(final AccountSto
         }
     }
 
+    private static void trackGutenbergEnabledForNonGutenbergSites(final SiteStore siteStore) {
+        for (SiteModel site : siteStore.getSites()) {
+            if (!TextUtils.equals(site.getMobileEditor(), GB_EDITOR_NAME)) {
```

Comment:
I think this is really just a theoretical concern, but _in theory_ if the dispatch call completed and updated the `site` before we got to this line, would it result in analytics _not_ being sent off because the site's mobile editor would have already been updated?

Reply:
Yeah, good point, better to move this before we dispatch the event.